Makefile

```diff
@@ -1,36 +1,53 @@
 # This Makefile contains common development / build commands for Package Analysis. For everything to work properly, it needs to be kept in the top-level project directory.
 
-REGISTRY := gcr.io/ossf-malware-analysis
-
+# Get absolute path to top-level package analysis project directory
 # outermost abspath removes the trailing slash from the directory path
 PREFIX := $(abspath $(dir $(abspath $(lastword $(MAKEFILE_LIST)))))
 SANDBOX_DIR := $(PREFIX)/sandboxes
 
-#
-# This is just the old 'build everything script'
-#
-.PHONY: legacy_build_docker
-legacy_build_docker:
-	bash build/build_docker.sh
+# Registry for Docker images built and used by package analysis
+REGISTRY := gcr.io/ossf-malware-analysis
+
+# If RELEASE_TAG environment variable is not specified, images will be tagged
+# as 'latest' which is equivalent to just tagging without specifying a version
+TAG := ${RELEASE_TAG}
+ifeq ($(TAG), )
+	TAG := latest
+	BUILD_ARG=
+else
+	# pass tag into analysis image build
+	BUILD_ARG=--build-arg=SANDBOX_IMAGE_TAG=$(TAG)
+endif
+
+# whether to push docker images
+DOCKER_PUSH := ${PUSH}
+ifeq ($(DOCKER_PUSH), )
+	DOCKER_PUSH := false
+endif
+
 
 .PHONY: all
 all: docker_build_all
 
 #
 # These recipes build all the top-level docker images
-# TODO grab release tag from env vars
 
 docker_build_%_image:
```

Comment:
Why not have a separate targets for pushing?

e.g.
```
docker_push_%_image: docker_build_%_image
  docker push --all-tags ${REGISTRY}$)(IMAGE_NAME)
```

Reply:
Either way is possible, but there are some things to consider

- having `docker_push_xyz` recipes mean there will be another 7ish sets of targets - one for for each concrete sandbox type
- I'm not sure if depending on the build target works because there's currently no way to mark the build as updated in Makefile. So running `make docker_build_xyz` and `make docker_push_xyz` might end up running the build twice. Can just remove that dependency though
- I was keeping the same environment variable format as the `build_docker.sh` script so it's easy to replace one with the other, but if that's not so important then I'm happy to not keep it the same

Reply:
@calebbrown I've implemented the extra recipes - see what you think